src/main/java/org/openstreetmap/atlas/geography/atlas/builder/proto/ProtoAtlasBuilder.java

```diff
@@ -0,0 +1,237 @@
+import org.openstreetmap.atlas.geography.Longitude;
+import org.openstreetmap.atlas.geography.PolyLine;
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.geography.atlas.AtlasMetaData;
+import org.openstreetmap.atlas.geography.atlas.builder.AtlasSize;
+import org.openstreetmap.atlas.geography.atlas.items.Line;
+import org.openstreetmap.atlas.geography.atlas.items.Point;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlas;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlasBuilder;
+import org.openstreetmap.atlas.proto.ProtoAtlasContainer;
+import org.openstreetmap.atlas.proto.ProtoLine;
+import org.openstreetmap.atlas.proto.ProtoLocation;
+import org.openstreetmap.atlas.proto.ProtoPoint;
+import org.openstreetmap.atlas.proto.ProtoTag;
+import org.openstreetmap.atlas.streaming.resource.Resource;
+import org.openstreetmap.atlas.streaming.resource.WritableResource;
+import org.openstreetmap.atlas.utilities.collections.Maps;
+import org.openstreetmap.atlas.utilities.collections.StringList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * Build an Atlas from a naive proto serialized file, or write an Atlas to a naive proto serialized
+ * file. At this stage, the builder can only handle Atlases that have tagged Points and Lines.
+ *
+ * @author lcram
+ */
+public class ProtoAtlasBuilder
+{
+    private static final Logger logger = LoggerFactory.getLogger(ProtoAtlasBuilder.class);
+
+    private static final String TAG_DELIMITER = "=";
+
+    /**
+     * Read a resource in naive ProtoAtlas format into a PackedAtlas.
+     *
+     * @param resource
+     *            the resource in naive ProtoAtlas format
+     * @return the constructed PackedAtlas
+     */
+    public PackedAtlas read(final Resource resource)
+    {
+        ProtoAtlasContainer protoAtlasContainer = null;
+
+        // First, we need to construct the container object from the proto binary
+        try
+        {
+            protoAtlasContainer = ProtoAtlasContainer.parseFrom(resource.readBytesAndClose());
+        }
+        // TODO do something smarter here?
+        catch (final InvalidProtocolBufferException exception)
+        {
+            throw new CoreException("Error parsing the protobuf");
+        }
+
+        // initialize atlas metadata
+        final long numberOfPoints = protoAtlasContainer.getNumberOfPoints();
+        final long numberOfLines = protoAtlasContainer.getNumberOfLines();
+        final long numberOfAreas = 0;
+        final long numberOfNodes = 0;
+        final long numberOfEdges = 0;
+        final long numberOfRelations = 0;
+        final AtlasSize atlasSize = new AtlasSize(numberOfEdges, numberOfNodes, numberOfAreas,
+                numberOfLines, numberOfPoints, numberOfRelations);
+        final AtlasMetaData atlasMetaData = new AtlasMetaData(atlasSize, true, "unknown",
```

Comment:
Wondering if we should have any kind of tag to specify this atlas was built from Proto?

Reply:
I'll take a look at that class and see if there's something I can add.

Reply:
Oh I just mean add an entry into the Metadata tags saying source=proto and proto_version=xx (not sure if you can read that dynamically, but assuming yes).

Reply:
Ah ok yeah can do. Did you have some fancy method in mind or should I just write code that greps build.gradle?

Reply:
I thought there might be an easier way! If there's not, I wouldn't worry about it.

Reply:
I'll mark it as TODO and if some simple way presents itself I'll be sure to implement it.